### ISSUE TITLE
build: Fix error handling in gulpfile so gulp serve doesn't break

### ISSUE THREAD

**Issue by ryanxcharles** [dattnetwork/datt#14]
There is a problem with error handling in the gulp file where when you run "gulp serve" if you get some kind of syntax error where browserify doesn't compile, browserify bundling the files. So we need to make the error handling work correctly so that when there is an error you can just fix the error and refresh the page without having to end gulp serve and rebuild everything.

**Comment by ryanxcharles** [comment]
Note that this was partially addressed here: https://github.com/dattnetwork/datt/commit/8587ed5c9912a2a9efcbc56492836002e443b835 ... but still not fixed.

**Comment by ryanxcharles** [comment]
I'm assigning this to myself since it annoys me to no end that the refreshing doesn't work correctly. I just did some tests and noticed this:

* Refreshing works as expected if you make a normal, non-broken change. I was able to change text on the page, and the page refreshed as expected.
* When I created a javascript syntax error, it showed the error in the CLI console. I then fixed the error. Things then refreshed as expected.
* When I changed the jsx to nonsense, that is the HTML embedded inside the javascript in the jsx files, this did not work as expected. There was a silent error that did not get spit out on the CLI console. The page refreshed, but it didn't show any changes. After I fixed the error, the page refreshed again, but still did not fix the errors.
